viewing_party/party.py

```diff
@@ -1,23 +1,185 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    #Create a data structure for each movie
+    movie = {}
+
+    #Check if title, genre, rating parameters are TRUE
+    if title and genre and rating != None:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie 
+
+    #Check if title, genre, rating parameters are FALSE
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    #Check if "watched" key exists:
+    if "watched" not in user_data:
+        #Set the "watched" key to the list of watched movies as a value
+        user_data["watched"] = []  
+    #Add movie to the watched list
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    #Add movie to watchlist inside of user_data
+    if "watchlist" not in user_data:
+        user_data["watchlist"] = movie
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    watched_movies = user_data["watched"]
+
+    for movie in watchlist:
+        if movie["title"] == title:
+            #Remove item of type dict from watchlist of type list
+            watchlist.remove(movie)
+            watched_movies.append(movie)
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data["watched"]
+    sum_rating = 0
+    count = 0
+    average_rating = 0.0
+    
+    #In case when watched_movies is an empty list
+    if watched_movies == []:
+        return average_rating
+    # Iterate through list of dicts called watched_movies
+    else:
+        for movie in watched_movies:
+            sum_rating += movie["rating"]
+            count += 1

```

Comment:
Like previously mention, I would remove the `else` and unindent the logic found within. That way we have less nested code and more readability.
```suggestion
    if watched_movies == []:
        return average_rating

    for movie in watched_movies:
        sum_rating += movie["rating"]
        count += 1

```